### PR TITLE
fix: [CDS-76206]: Add infra selectors in shell script step

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
When executing shell script if the script contains the string HARNESS_KUBE_CONFIG_PATH or the user selected the checkbox to Include Infrastructure Selectors and deployment type is Kubernetes, then a delegate will be selected according to the k8s infrastructure selected for the pipeline execution.
Tested my changes with adding includeInfraSelectors: true to pipeline yaml and delegate was selected from k8s infra.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51653)

### CODE REVIEW

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51653)* status: 10 file(s) to review. 1 of 2 approvals obtained.

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/utils/K8sUtils.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import io.harness.delegate.task.k8s.RancherK8sInfraDelegateConfig;
+import io.harness.expression.ExpressionEvaluator;
+
+import java.util.List;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class K8sUtils {
+  public static void getExecutionCapabilities(ExpressionEvaluator maskingEvaluator,
```
Comment (yogesh-chauhan) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/utils/K8sUtils.java:30]: 2 things here
- better naming of class maybe K8sCapabilityGenerate
- method should not modify argument. It should return List<ExecutionCapability> capabilities

**Review by yogesh-chauhan — COMMENTED** [review]
I think you need to update schema file also

**Comment by sarthakkasat** [comment]
trigger uts

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51653)* status: 15 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/utils/K8sUtils.java` line 30 at r4](https://reviewable.io/reviews/harness/harness-core/51653#gh-1304221200:-NccI3l70Yq3xj8f3kuN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/aa046d771c2bfd2f89bcffeecd66eed18bc6241c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/utils/K8sUtils.java#L30)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

2 things here
- better naming of class maybe K8sCapabilityGenerate
- method should not modify argument. It should return List<ExecutionCapability> capabilities
</blockquote></details>

Done.

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 5 of 8 files at r1, 6 of 7 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 4 of 4 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51653)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Comment by sarthakkasat** [comment]
trigger uts

**Comment by sarthakkasat** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51653)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 8 files at r1, 3 of 7 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 1 of 4 files at r9, 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51653)* status: 1 file(s) to review.

**Comment by sarthakkasat** [comment]
trigger review

**Pull request merged**